lib/client/keystore_test.go

```diff
@@ -80,6 +81,29 @@ func TestListKeys(t *testing.T) {
 	require.Equal(t, samKey.Pub, skey.Pub)
 }
 
+func TestEmbeddedDirsInKeystoreIsNotAnError(t *testing.T) {
+	s, cleanup := newTest(t)
+	defer cleanup()
+
+	// Given a key store with a valid directory structure
+	host := "some-host"
+	user := "zaphod"
+	key := s.makeSignedKey(t, false)
+	require.NoError(t, s.addKey(host, user, key))
+
+	// ... *and* an invalid folder injected into it
+	p, err := s.store.dirFor(host, false)
+	require.NoError(t, err)
+	spuriousDirPath := path.Join(p, user+"-db", "root", "aaa-should-not-be-here")
+	require.NoError(t, os.MkdirAll(spuriousDirPath, 0700))
+
+	// When I attempt to enumerate the DB keys
```

Comment:
nit:
```suggestion
	// When attempting to enumerate the DB keys
```

Reply:
Why? (Not trying to be argumentative, just want to understand the rationale as I've always seen Given/When/Expect test info written in the first person)